src/RobotsTxt.php

```diff
@@ -89,37 +130,44 @@ protected function getDisallowsPerUserAgent(string $content): array
                 continue;
             }
 
-            $disallowUrl = $this->parseDisallow($line);
+            $rule = null;
+            $disallowUrl = $this->parse($line, $rule);
 
-            $currentUserAgent[$disallowUrl] = $disallowUrl;
+            if ($disallowUrl !== null) { // other than allow/disallow
+                $currentUserAgent[$disallowUrl] = $rule;
+            }
         }
 
         return $disallowsPerUserAgent;
     }
 
-    protected function isCommentLine(string $line): bool
-    {
-        return strpos(trim($line), '#') === 0;
-    }
-
     protected function isUserAgentLine(string $line): bool
     {
-        return strpos(trim($line), 'User-agent') === 0;
+        return preg_match('/^User-agent\s*:/i', trim($line));
```

Comment:
Don't use a regex here. It's more readable to just call `strtolower` on `$line` and compare that to `user-agent`.

Reply:
Done.